lib/index.js

```diff
@@ -1,217 +1,354 @@
-const spauth = require('node-sp-auth')
+const process = require('node:process')
+const crypto = require('node:crypto')
+const fs = require('node:fs')
+const msal = require('@azure/msal-node')
 const axios = require('axios')
+const { v4: uuid } = require('uuid')
 
-const getGUID = () => {
-  let d = Date.now()
-  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
-    const r = (d + Math.random() * 16) % 16 | 0
-    d = Math.floor(d / 16)
-    return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16)
-  })
-}
-
+// see https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/get-to-know-the-sharepoint-rest-service?tabs=csom
 class Sharepoint {
-  constructor (url) {
-    if (!url) {
-      throw new Error('You must provide a url.')
+  /**

```

Comment:
Can this comment be removed? It's already in the readme and just pollutes the code being in here too

Reply:
Tabled of env vars removed.